tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -193,23 +262,34 @@ static int spi_null_tx_buf(struct device *dev, struct spi_config *spi_conf)
 		return -1;
 	}
 
-	LOG_INF("Passed");
+	LOG_INF("Passed null rx");
 
 	return 0;
 }
+#endif /* CONFIG_SPI_STM32_DMA */
 
 static int spi_rx_half_start(struct device *dev, struct spi_config *spi_conf)
 {
 	const struct spi_buf tx_bufs[] = {
 		{
 			.buf = buffer_tx,
+#ifdef CONFIG_SPI_STM32_DMA
+	/* transfer half buffer, has TH irq not enabled */
+			.len = 8,
+#else
 			.len = BUF_SIZE,
+#endif
 		},
 	};
 	const struct spi_buf rx_bufs[] = {
 		{
 			.buf = buffer_rx,
+#ifdef CONFIG_SPI_STM32_DMA
+	/* transfer half buffer, has TH irq not enabled */
+			.len = BUF_SIZE,
+#else
 			.len = 8,
+#endif
```

Comment:
why inverting the lengths here? And why is that stm32 dma specific as well?

Reply:
spi_null_rx_buf removed